src/java.net.http/share/classes/jdk/internal/net/http/Http1Response.java

```diff
@@ -275,13 +277,23 @@ long fixupContentLen(long clen) {
      * Read up to MAX_IGNORE bytes discarding
      */
     public CompletableFuture<Void> ignoreBody(Executor executor) {
-        int clen = (int)headers.firstValueAsLong("Content-Length").orElse(-1);
-        if (clen == -1 || clen > MAX_IGNORE) {
+
+        // Read the `Content-Length` header
+        long clen;
+        try {
+            clen = readContentLength(headers, "", 0);
```

Comment:
This does not look right - we should keep -1 as meaning that content length was not found.